### DISCUSSION TITLE
如何支持Windows XP系统、配置文件中的ID和secret加密

### DISCUSSION THREAD

**Discussion by fonxaohu** [jeessy2/ddns-go#1071] [Q&A]
1. 如何支持Windows XP系统？下载Windows I386版本，并不支持WindowsXP Windows2003，但是有不少工作电脑必须是这个系统。有什么办法支持吗
2. 配置文件中的ID和secret建议能加密，以免泄漏重要信息

**Comment by WaterLemons2k** [comment]
> 1. 如何支持Windows XP系统？下载Windows I386版本，并不支持WindowsXP Windows2003，但是有不少工作电脑必须是这个系统。有什么办法支持吗

Windows XP 最高只支持到 Go 1.10，故无法支持。（https://go.dev/doc/go1.10#ports ）

> 2. 配置文件中的ID和secret建议能加密，以免泄漏重要信息

- #273

**Reply by fonxaohu** [reply]
混淆加密不支持腾讯云阿里云吗

**Reply by WaterLemons2k** [reply]
> 混淆加密不支持腾讯云阿里云吗

目前只有前端会混淆加密；除了 Callback 都支持。

https://github.com/jeessy2/ddns-go/blob/8474a5a1563643c52df176a31f4114d23efa61a6/web/writing.go#L84-L85

**Reply by fonxaohu** [reply]
您的意思是说：.yaml配置文件不能加密ID、Secret吗

**Reply by WaterLemons2k** [reply]
> 您的意思是说：.yaml配置文件不能加密ID、Secret吗

不能，请参考 https://github.com/jeessy2/ddns-go/pull/273#issuecomment-1127173863

**Comment by fonxaohu** [comment]
不支持用Go 1.10编译为支持XP 的32位exe吗

**Reply by WaterLemons2k** [reply]
用了很多不支持 Go 1.10 的库，所以不支持用 Go 1.10 编译。最低支持版本请见 [go.mod](../blob/master/go.mod)
